core/arch/riscv/include/riscv.h

```diff
@@ -77,6 +77,30 @@
 		__tmp;							\
 	})
 
+#define readable_csr(csr)						\
+	({								\
+		bool __readable = true;					\
+		unsigned long __tmp = 0;				\
+		asm volatile ("	j 1f\n"					\
+			"	.align 4\n"				\
+			"2:\n"						\
+			"	csrr %0, sepc\n"			\
+			"	addi %0, %0, 4\n"			\
+			"	csrw sepc, %0\n"			\
+			"	add %0, x0, x0\n"			\
+			"	sret\n"					\
```

Comment:
I guess this is a return instruction, that's bound to surprise other users of this macro.

Reply:
Agreed, let me fix it.